graph-builder/src/registry.rs

```diff
@@ -123,23 +254,13 @@ fn fetch_metadata(registry: &str, repo: &str, tag: &str) -> Result<release::Meta
     bail!("metadata document not found in image")
 }
 
-fn fetch_metadata_from_layer(
-    base: &Url,
-    repo: &str,
-    layer: &Layer,
+fn extract_metadata_from_layer_blob(
+    blob: &dkregistry::v2::blobs::Blob,
+    metadata_filename: &str,
 ) -> Result<release::Metadata, Error> {
-    trace!("fetching metadata from {}", layer.blob_sum);
-
-    let response = reqwest::get(base.join(&format!("v2/{}/blobs/{}", repo, layer.blob_sum))?)
-        .context("failed to fetch image blob")?;
+    trace!("Looking for {} in archive", metadata_filename);
 
-    ensure!(
-        response.status().is_success(),
-        "failed to fetch metadata document: {}",
-        response.status()
-    );
-
-    let mut archive = Archive::new(GzDecoder::new(response));
+    let mut archive = Archive::new(GzDecoder::new(&blob[..]));
```

Comment:
@crawford @lucab can one of you confirm that this doesn't unnecessarily copy the whole blob?

Reply:
I think it doesn't. You can also use a `Cursor` here.

Reply:
I'm not sure what `blob` is, but that usually won't make a copy. You are asking for a slice with `[..]`, but because the size of a slice isn't known at compilation, you need to take a reference to it with `&`.

Reply:
Here `blob` is a `Vec<u8>` which doesn't implement `Read` itself so I made it into a slice.

Reply:
In that case, you can do

```rust
    let mut archive = Archive::new(GzDecoder::new(blob.as_ref()));
```